internal/trie/node/hash_test.go

```diff
@@ -119,7 +119,7 @@ func Test_Branch_EncodeAndHash(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			t.Parallel()
 
-			encoding, hash, err := testCase.branch.EncodeAndHash()
+			encoding, hash, err := testCase.branch.EncodeAndHash(false)
```

Comment:
please add `isRoot` as a field to the test case. And also add a test case with `isRoot: true`.